File: src/main/java/com/mmust/evaluators/impl/ShuntingYardInfixToPostfixParser.java
package com.mmust.evaluators.impl;

import com.mmust.token.Token;
import com.mmust.evaluators.InfixToPostfixParser;

public class ShuntingYardInfixToPostfixParser implements InfixToPostfixParser {

	@Override
	public Iterable<Token> parse(Iterable<Token> infixExpression) {
		ShuntingYard shuntingYard = new ShuntingYard(infixExpression);
		return shuntingYard.doSorting();
	}
}

File: src/main/kotlin/com/mmust/evaluators/impl/ShuntingYardInfixToPostfixParser.kt
package com.mmust.evaluators.impl

import com.mmust.evaluators.InfixToPostfixParser
import com.mmust.token.Token

class ShuntingYardInfixToPostfixParser : InfixToPostfixParser {

    override fun parse(infixExpression: Iterable<Token>): Iterable<Token> {
        return ShuntingYard(infixExpression).doSorting()
    }
}
